moviebot/database/database.py

```diff
@@ -13,76 +16,88 @@
 
 
 class DataBase:
-    """DataBase class for SQL databases. It provides the functionality
-    to search the database according to user preferences."""
+    """DataBase class for SQL databases.
+
+    It provides the functionality to search the database according to
+    user preferences.
+    """
 
     def __init__(self, path):
-        """Initializes the internal structures of the DataBase class
+        """Initializes the internal structures of the DataBase class.
 
         Args:
-            path: path to the database file
-
+            path: Path to the database file.
         """
         self.db_file_path = path
-        self.sql_connection = sqlite3.connect(
-            self.db_file_path
-        )  # SQL connection required to
-        # access the database
-        self.db_table_name = self._get_table_name()  # name of the table
+        # self.sql_connection = sqlite3.connect(
+        #     self.db_file_path
+        # )  # SQL connection required to
+        # # access the database
+        # self.db_table_name = self._get_table_name()  # name of the table
+        self._initialize_sql()
         self.current_CIN = None
         self.backup_db_results = None
 
-    def database_lookup(self, dialogue_state, ontology):
-        """Performs an SQL query to answer a user requirement.
+    def _initialize_sql(self):
+        """Initializes the SQL connection and the name of the table to query."""
+        self.sql_connection = sqlite3.connect(self.db_file_path)
+        self.db_table_name = self._get_table_name()
+
+    def get_sql_condition(
+        self, dialogue_state: DialogueState, ontology: Ontology
+    ) -> Union[str, None]:
+        """Returns the condition for a SQL query based on dialogue state.
 
         Args:
-            dialogue_state: the current dialogue state
-            ontology: ontology to check specific parameters
+            dialogue_state: Dialogue state.
+            ontology: Ontology to check specific parameters.
 
         Returns:
-            the results of the SQL query
+            SQL condition if possible.
+        """
+        args = []
+        if dialogue_state.agent_should_offer_similar:
+            similar_movies = list(dialogue_state.similar_movies.values())[0]
+            if len(similar_movies) > 0:
+                for title in similar_movies:
+                    args.append(f'{Slots.TITLE.value} = "{title}"')
+                return " OR ".join(args) if len(args) > 0 else None
```

Comment:
I am unsure about the implications here, but if `args` is empty `database_lookup` should return an empty list.

Reply:
Good catch! I changed a bit the method `get_sql_condition` and added a if in the `database_lookup` to return an empty list if no similar movies were found.